### ISSUE TITLE
Segmentation fault with the latest version of Prosody and Lua 5.2

### ISSUE THREAD

**Issue by sbrl** [hoelzro/lua-proctitle#1]
As the title suggests. As soon as I build & install this module with the latest version of this module on Ubuntu Server 18.04 with the latest version of Prosody from the apt repository and restart Prosody, I get a segmentation fault.

**Comment by hoelzro** [comment]
@sbrl Thanks for the report!  Did you build this module against Lua 5.2, or Lua 5.3?  Can you reproduce the segmentation fault outside of Prosody with a simple Lua script?

**Comment by sbrl** [comment]
@hoelzro I'm unsure. The default version of Lua on my server is 5.2.

Unfortunately, I'm rather inexperienced with Lua. How would I go about testing it in the way you describe?

**Comment by hoelzro** [comment]
@sbrl Looking at proctitle's makefile, it seems that the module always builds against Lua 5.1 - could you change `-llua5.1` to `-llua5.2` on line 2, and `-I/usr/include/lua5.1` to `-I/usr/include/lua5.2` on line 5, and see if that fixes the issue?

**Comment by sbrl** [comment]
Yep, that seems to work - although I need to install the `liblua5.2-dev` package, apparently.

**Comment by hoelzro** [comment]
Ah ha - that makes sense!  If everything's working, do you mind if I close this issue?

**Comment by sbrl** [comment]
Erm can do. Was this module written for Prosody? If so, then this should probably be updated, as the currently latest stable of Prosody runs on Lua 5.2.

At any rate, it's really not intuitive which version of Lua it's building against - or, indeed how to build it, or how to install it once it's built (I guessed based on the error from Prosody).

Perhaps some better documentation (or a `luarocks` package? Not sure what that is / how that works) would help there.

**Comment by hoelzro** [comment]
@sbrl Those are all good points - I wrote this for myself, but I definitely agree it's not super clear how to use it.  I'll add some more flexibility, some instructions, and maybe a luarock when I have some time!
